pandas/core/indexes/base.py

```diff
@@ -415,6 +415,11 @@ def asi8(self):
         ndarray
             An ndarray with int64 dtype.
         """
+        warnings.warn(
+            "Index.asi8 is deprecated and will be removed in a future version",
```

Comment:
can you add that the replacement is `.astype('i8')`

Reply:
trouble is thats not really correct for the Index classes we're deprecating it for.  For these classes .asi8 returns None

Reply:
ahh ok. are we deprecating for DTI and cousins? (I think we should do that as well to be consistent).

Reply:
ill take a look; that might be pretty invasive

Reply:
ok i think you should make this an internal method and deprecate the user facing one (even though we use it internally in the many parts of pandas)